### DISCUSSION TITLE
Need more permissions in current SP API App

### DISCUSSION THREAD

**Discussion by bpyzikvc** [amzn/selling-partner-api-models#3065] [General] [answered]
I have an app that has over 100 users on it already. I now have the need to use the get product listing api and am getting a forbidden response. I assume this is because in the settings for my app, product listing permission is unchecked. If I check this, will this affect any of the refresh tokens I already have for my users? Will they immediately be able to use the new functionality or will they need to reauthorize my app to get a new refresh token? 

Please advise, thank you!

**Comment by sungolivia** [comment]
when you say permissions, are you referring to [roles](https://developer-docs.amazon.com/sp-api/docs/roles-in-the-selling-partner-api)?

If so, if you apply for a new role for through your developer profile, you will need to wait for an approval, then you can apply the new role to your app config. Once you submit the app for re-listing, re-auths will be required from your users. 

So not right away unless you have the role already approved and applied to your app!

Check out the blog for screenshots of what I am talking about! https://developer-docs.amazon.com/sp-api-blog/docs/choosing-roles-for-developer-profile-and-sp-api-app

**Comment by bpyzikvc** [comment]
@sungolivia Yes I do mean roles. However, my developer profile already has the new role applied. It is my app that does not. So would changing the app to also have the role selected also require re-auths?

I also want to note that when a user authorizes they already are presented with the screen that says we have the product listing role, even when the app does not.

Thanks for your response!

**Comment by sungolivia** [comment, marked as answer]
is your app hybrid? If it was hybrid, is it fully migrated to pure SP-API app? If you can tell me more about your app setup, it would help!

But to provide more context, if your app didn't have product listing role and now you are adding this role and publish the app, your users will need to re-auth since you are expanding the consent scope from your users.

**Reply by bpyzikvc** [reply]
@sungolivia Thank you for this answer, we were hoping to get some clarification about the re-auth our users would need to do. 
Does this mean if we added the role, does all functionality of the app break until the user reauthorizes? 
Or (hopefully) they only need to reauthorize if they intend to use the products functionality?

**Reply by sungolivia** [reply]
nono the functionality of the app before adding the role should all be there even after adding the role before getting re-auth. But until you get re-auth for the new role that's added, you won't be able to access whatever that the new role provides.
Does that clarify a bit?

**Reply by bpyzikvc** [reply]
Yes! sorry for late response, we did not want to make any changes during busy season but this makes it crystal clear for us. Thank you

**Comment by bpyzikvc** [comment]
@sungolivia No it is a pure SP-API app. We set it up months ago but now would like to also have access to the get product apis. Not sure what other info you are looking for.

Thank you for that answer about reauthorizing though, that solves that. I think instead we will use a product report since those seem to work with or without the role applied. The api would have been a nice to have
